File: src/main/java/dev/elysium/eraces/bootstrap/ListenerInitializer.java
package dev.elysium.eraces.bootstrap;

import dev.elysium.eraces.ERaces;
import dev.elysium.eraces.exceptions.internal.InitFailedException;
import dev.elysium.eraces.listeners.GuiListener;
import dev.elysium.eraces.listeners.RaceSelectListener;
import dev.elysium.eraces.listeners.PluginMessageListener;
import org.bukkit.Bukkit;

public class ListenerInitializer implements IInitializer {
    @Override
    public void setup(ERaces plugin) {
        try {
            Bukkit.getPluginManager().registerEvents(RaceSelectListener.INSTANCE, plugin);
            plugin.getServer().getMessenger().registerIncomingPluginChannel(plugin, "elysium:eraces_cast", new PluginMessageListener());

            GuiListener.INSTANCE.init(plugin);
            Bukkit.getPluginManager().registerEvents(GuiListener.INSTANCE, plugin);
        } catch (Exception e) {
            throw new InitFailedException("Ошибка при регистрации слушателей", e);
        }
    }
}

File: src/main/java/dev/elysium/eraces/bootstrap/ListenerInitializer.kt
package dev.elysium.eraces.bootstrap

import dev.elysium.eraces.ERaces
import dev.elysium.eraces.exceptions.internal.InitFailedException
import dev.elysium.eraces.listeners.GuiListener
import dev.elysium.eraces.listeners.GuiListener.init
import dev.elysium.eraces.listeners.PluginMessageListener
import dev.elysium.eraces.listeners.RaceSelectListener
import org.bukkit.Bukkit

class ListenerInitializer : IInitializer {
    override fun setup(plugin: ERaces) {
        try {
            Bukkit.getPluginManager().registerEvents(RaceSelectListener, plugin)
            plugin.server.messenger
                .registerIncomingPluginChannel(plugin, "elysium:eraces_cast", PluginMessageListener())

            init(plugin)
            Bukkit.getPluginManager().registerEvents(GuiListener, plugin)
        } catch (e: Exception) {
            throw InitFailedException("Ошибка при регистрации слушателей", e)
        }
    }
}
